Compile/src/Producers/PrefiringWeightProducer.cc

```diff
@@ -1,162 +1,338 @@
-            if (nonprefiringprobfromoverlappingphotons == 1.)
-            {
-                nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet;
+        }
+        //Now calculate prefiring weights for muons
+        if (!missingInputMuon_) {
+            for (KMuons::iterator muon = theMuons->begin(); muon != theMuons->end(); ++muon) {
+                double pt = muon->p4.Pt();
+                double phi = muon->p4.Phi();
+                double eta = muon->p4.Eta();
+                // Remove crappy tracker muons which would not have prefired the L1 trigger
+                if (pt < 5 || !muon->idLoose())
+                    continue;
+                double prefiringprob_mu = getPrefiringRateMuon(eta, phi, pt, fluct);
+                nonPrefiringProbaMuon[fluct] *= (1. - prefiringprob_mu);
             }
-            //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
-            else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet)
-            {
-                if (nonprefiringprobfromoverlappingphotons != 0.)
-                {
-                    nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
-                } else
-                {
-                    nonPrefiringProba[fluct] = 0.;
-                }
+        }
+    }
+    // Calculate combined weight as product of the weight for individual objects
+    for (const auto fluct : {fluctuations::central, fluctuations::up, fluctuations::down}) {
+        nonPrefiringProba[fluct] = nonPrefiringProbaECAL[fluct] * nonPrefiringProbaMuon[fluct];
+    }
+    // Calculate statistical and systematic uncertainty separately in the muon case
+    for (const auto fluct :
+         {fluctuations::upSyst, fluctuations::downSyst, fluctuations::upStat, fluctuations::downStat}) {
+        if (!missingInputMuon_) {
+            for (KMuons::iterator muon = theMuons->begin(); muon != theMuons->end(); ++muon) {
+                double pt = muon->p4.Pt();
+                double phi = muon->p4.Phi();
+                double eta = muon->p4.Eta();
+                // Remove crappy tracker muons which would not have prefired the L1 trigger
+                if (pt < 5 || !muon->idLoose())
+                    continue;
+                double prefiringprob_mu = getPrefiringRateMuon(eta, phi, pt, fluct);
+                nonPrefiringProbaMuon[fluct] *= (1. - prefiringprob_mu);
             }
-            //Last case: if overlapping photons have a non prefiring rate smaller than the jet, don't consider the jet in the event weight, and do nothing.
         }
     }
+    //Move global prefire weights, as well as those for muons, photons, and jets, to the event
+    //Up and Down are switched, since we use the inverse and apply it to data.
     product.m_optionalWeights["prefiringWeight"] =  1./nonPrefiringProba[0];
```

Comment:
can this be 1/0?

Reply:
Default value is 1. And if we have a missing input the calculation s skipped. So default is still 1.

Reply:
fine.